### ISSUE TITLE
missing generic type in .fromJson

### ISSUE THREAD

**Issue by smiLLe** [rrousselGit/freezed#32]
after creating these model classes on version 0.2.4...
```dart
/// model.dart

import 'package:json_annotation/json_annotation.dart';
import 'package:flutter/foundation.dart';

part 'model.freezed.dart';
part 'model.g.dart';

@immutable
abstract class Parent<T> with _$Parent<T> {
  const factory Parent(@DataConverter() List<T> items) = _Parent<T>;

  factory Parent.fromJson(Map<String, dynamic> json) =>
      _$ParentFromJson<T>(json);
}

@immutable
abstract class Item with _$Item {
  const factory Item(String name) = _Item;

  factory Item.fromJson(Map<String, dynamic> json) => _$ItemFromJson(json);
}

class DataConverter<T> implements JsonConverter<T, Object> {
  const DataConverter();

  @override
  T fromJson(Object json) {
    return Item.fromJson(json) as T;
  }

  @override
  Object toJson(T object) {
    return object;
  }
}
```
... i am going to run the default sample project. ( flutter create ). It will result in the following error:
```dart
lib/model.freezed.dart:74:57: Error: The constructor function type '_$_Parent<dynamic> Function(Map<String, dynamic>)' isn't a subtype of '_Parent<T> Function(Map<String, dynamic>)'.
 - '_$_Parent' is from 'package:freezed_test/model.dart' ('lib/model.dart').
 - 'Map' is from 'dart:core'.
 - '_Parent' is from 'package:freezed_test/model.dart' ('lib/model.dart').
  factory _Parent.fromJson(Map<String, dynamic> json) = _$_Parent.fromJson;
```

I can fix the error by opening model.freezed and change

```dart
/// model.freezed.dart:74

factory _Parent.fromJson(Map<String, dynamic> json) = _$_Parent.fromJson;
to
factory _Parent.fromJson(Map<String, dynamic> json) = _$_Parent<T>.fromJson;
```

**Comment by rrousselGit** [comment]
Fixed in 0.2.5 😄

**Comment by smiLLe** [comment]
I will try it immediately :) Thanks for the fast response!
